integration/simulation/l2_network.go

```diff
@@ -27,21 +30,30 @@ func NewL2Network(avgBlockDuration uint64, avgLatency uint64) *L2NetworkCfg {
 	}
 }
 
-// BroadcastRollup Broadcasts the rollup to all L2 peers
-func (cfg *L2NetworkCfg) BroadcastRollup(r obscurocommon.EncodedRollup) {
-	for _, a := range cfg.nodes {
-		rol := nodecommon.DecodeRollupOrPanic(r)
-		if a.ID != rol.Header.Agg {
-			t := a
-			obscurocommon.Schedule(cfg.delay(), func() { t.P2PGossipRollup(r) })
-		}
+func (cfg *L2NetworkCfg) BroadcastTx(tx nodecommon.EncryptedTx) {
+	for _, a := range cfg.nodeTxAddresses {
+		address := a
+		obscurocommon.Schedule(cfg.delay()/2, func() {
```

Comment:
I'd suggest playing with the delay setting, and making it much lower, because there will be a delay now anyway.

Actually, my suggestion is to first measure how much is us it takes to send a rollup from one node to another and then make the block duration around 25 times that,and then add an average latency of avgBlockDurationUSecs / 25.
(these are just finger in the air)

Reply:
@tudor-malene Thanks, done. That allowed me to revert to the original efficiency levels. Note that I don't inject any latency for the rollups sent around by the node, just the ones pushed out by the l2_network.go.